source/compiler.ts

```diff
@@ -8,13 +8,24 @@ import * as ts from "typescript";
 // There is an example that uses the LanguageService here:
 // https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API#incremental-build-support-using-the-language-services
 
+// prettier-ignore
+export type CompilerOptions = {
+  [K in keyof ts.CompilerOptions]: ts.CompilerOptions[K] | (
+    K extends 'moduleResolution' ? 'node'|'classic' :
+    K extends 'target' ? keyof typeof ts.ScriptTarget | (
+      'es3'|'es5'|'es6'|'es2015'|'es2016'|'es2017'|'es2018'|'es2019'|'es2020'|'esnext'
+    ) :
+    void
+  )
+}
+
```

Comment:
I'm fine with using the `CompilerOptions` type from `typescript`, but I do not think we should be 'fixing' it here. If there are option values that are missing, this is something that should be addressed in the TypeScript repo.
 Basically, if there would be reasons for their not accepting a PR that adds the 'missing' values to TypeScript's `CompilerOptions`, I think those reasons would apply here, too.

Reply:
I can submit a PR in TS. I did this because I saw tests that use some of these values, like "es2015" and "node", and these are not in CompilerOptions, so that broke the tests and I decided to patch it. I'm pretty sure TS supports these values so it's weird that they are not included in the type.

Reply:
Actually, I don't think we can make this change. The options we are talking about here are the options that are passed to `convertCompilerOptionsFromJson` and they are typed as `any` in the TS API. The `CompilerOptions` are what's *returned* by that function.

Reply:
Thing is, these unsupported values work fine - I tried compiling a snippet that uses flatMap with target set to 'es2019', but Microsoft might not have enough incentive to extend their types and keep it simple instead. But if it's not extended in this PR on the other hand, it might break existing and perfectly valid code.

Reply:
The problem is that the `CompilerOptions` type refers to what's returned from `convertCompilerOptionsFromJson` - which is called in the ctor - and not to what's accepted by the ctor. E.g. the `target` in `CompilerOptions` is this enum:

<img width="210" alt="target" src="https://user-images.githubusercontent.com/3878593/88857597-f1356a80-d239-11ea-9d3b-9947303bfd15.PNG">

And the numeric values are [not shown](https://www.typescriptlang.org/docs/handbook/compiler-options.html) as valid values JSON options - which are what the options passed to the ctor are, as the intention was for what's passed to map directly to what would be in a `tsconfig.json` file.

Reply:
Ah yeah you're right.